plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -163,16 +163,18 @@ private List<BigQuerySplit> readFromBigQuery(ConnectorSession session, TableDefi
     }
 
     @VisibleForTesting
-    ReadSession createReadSession(ConnectorSession session, TableId remoteTableId, List<String> projectedColumnsNames, Optional<String> filter, int actualParallelism)
+    ReadSession createReadSession(ConnectorSession session, TableId remoteTableId, List<String> projectedColumnsNames, Optional<String> filter)
     {
         ReadSessionCreator readSessionCreator = new ReadSessionCreator(bigQueryClientFactory, bigQueryReadClientFactory, viewEnabled, arrowSerializationEnabled, viewExpiration, maxReadRowsRetries);
-        return readSessionCreator.create(session, remoteTableId, projectedColumnsNames, filter, actualParallelism);
+        // At least 100 to cater for cluster scale up
+        int desiredParallelism = Math.min(nodeManager.getRequiredWorkerNodes().size() * 3, 100);
```

Comment:
Can this cause issues on BigQuery API ? 

https://github.com/trinodb/trino/blob/30104c7247daf15d2ab0c9b00f9d9b3009b9becb/docs/src/main/sphinx/connector/bigquery.md?plain=1#L85-L91

Reply:
the existing docs are inaccurate. We just create one read stream per worker today.
After this change we'll create `min(worker * 3, 100)`.

I worry about running into BigQuery quotas (https://cloud.google.com/bigquery/quotas#storage-limits).

The most relevant being 2000 connections (for some regions), 400 connections for most regions (which means at most 4 concurrent queries in those regions) for each bigquery catalog.

Reply:
Also is there any point to requesting more than `task.concurrency * nodes` streams? IIRC we don't multiplex table scan splits so others would just wait in line.

Reply:
number of worker nodes may change over time

Reply:
I expected to see a `clamp` call here, so you have a minimum (say 20), mid (say 5 * node), max (say 100).

Reply:
@dain See https://github.com/trinodb/trino/pull/22279#discussion_r1629366672

So maybe the comment / variable should be made more clear to indicate BigQuery can give less if it wants. 😄

Reply:
yes, i will refactor the code, moving the logic closer to the usage